### PR TITLE
[PVT-182502615] JSON logging

### PR DESCRIPTION
**Pull request by andyzito**
This uses Lograge to enable STDOUT (Docker container log) JSON logging for request/controller events, and a simple custom formatter for custom log events (`Rails.logger.debug/info/etc...`). Not deployed anywhere yet. The tests I ran indicate that request logs and custom log events are both correctly piped to the Docker log as raw JSON. We will need to handle exception notification separately, Lograge specifically discourages users from using it for exception logging/notification.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -283,4 +283,12 @@ def prepare_exception_notifier
       current_user_browser: browser.to_s,
     }
   end
+
+  # For Lograge
+  def append_info_to_payload(payload)
+    super
+    payload[:host] = request.host
+    payload[:remote_ip] = request.remote_ip
```
Comment (eanders) [inline app/controllers/application_controller.rb]: This is already in the same file on line 133.  I think you could just add 
`payload[:ip]` up there.

File: config/environments/development.rb [hunk 2]
```diff
@@ -131,6 +129,29 @@
   routes.default_url_options ||= {}
   routes.default_url_options[:script_name]= ''
 
+  # https://tasdikrahman.me/2020/07/07/structured-logging-in-rails/
+  config.lograge.formatter = Lograge::Formatters::Json.new
+  config.lograge.enabled = true
+  config.lograge.base_controller_class = ['ActionController::Base']
+  config.lograge.custom_options = lambda do |event|
+    {
+      request_time: Time.now,
```
Comment (eanders) [inline config/environments/development.rb]: maybe `Time.current`?

File: config/environments/development.rb [hunk 3]
```diff
@@ -131,6 +129,29 @@
...
+      application: Rails.application.class,
+      process_id: Process.pid,
+      host: event.payload[:host],
+      remote_ip: event.payload[:remote_ip],
```
Comment (eanders) [inline config/environments/development.rb]: Much of this is already in `config/application.rb` maybe we should centralize it all there?

**Review by eanders — COMMENTED** [review]
This looks reasonable, but I think we already have places for both of these changes, so let's centralize if possible.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (andyzito) [reply app/controllers/application_controller.rb]: 👍

File: config/environments/development.rb (hunk 3 above)
Comment (andyzito) [reply config/environments/development.rb]: 👍

File: config/environments/development.rb (hunk 2 above)
Comment (andyzito) [reply config/environments/development.rb]: 👍

**Comment by andyzito** [comment]
I made a change to explicitly point Lograge at STDOUT, and the logs are now flowing into Grafana/Loki. If @eanders can give me a hand testing custom log events on Monday, we might be closer to done (maybe).

**Comment by andyzito** [comment]
@eanders Bumping this. If you can go into the console and web interface and trigger a custom logging event or two, then we can confirm that those flow into Grafana.

**Comment by andyzito** [comment]
The following is working on three warehouse staging:
- Request logs, in JSON, flowing to Grafana
- Custom logging events triggered in-application, in JSON, flowing to Grafana

And the following is not covered here:
- Custom logging events triggered in a console session
- Exception handling (still covered by our slack notifier system for now)

**Review by eanders — APPROVED** [review]
This looks good to me.  Should we merge now, or are we waiting for more infrastructure?

**Comment by andyzito** [comment]
Nothing we're waiting for that I know of, @schwartzkylej ?

**Comment by schwartzkylej** [comment]
I'm good with it.

**Pull request merged**